motoman_driver/src/io_relay.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ * Copyright (c) 2016, Delft Robotics Institute
+ * Copyright (c) 2020, Southwest Research Institute
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright
+ *  notice, this list of conditions and the following disclaimer.
+ *  * Redistributions in binary form must reproduce the above copyright
+ *  notice, this list of conditions and the following disclaimer in the
+ *  documentation and/or other materials provided with the distribution.
+ *  * Neither the name of the Southwest Research Institute, nor the names
+ *  of its contributors may be used to endorse or promote products derived
+ *  from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "motoman_driver/io_relay.h"
+#include <string>
+#include <ros/ros.h>
+
+namespace motoman
+{
+namespace io_relay
+{
+
+using industrial::shared_types::shared_int;
+
+bool MotomanIORelay::init(int default_port)
+{
+  ROS_DEBUG_NAMED("init", "MotomanIORelay::init(default_port = %d)", default_port);
+
+  std::string ip;
+  int port;
+
+  ros::param::get("robot_ip_address", ip);
+  // override port with ROS param, if available
+  ros::param::param<int>("~port", port, default_port);
+
+  // check for valid parameter values
+  if (ip.empty())
+  {
+    ROS_ERROR_NAMED("init", "No valid robot IP address found.  Please set ROS 'robot_ip_address' param");
+    return false;
+  }
```

Comment:
```suggestion
  // override port with ROS param, if available
  const std::string port_parameter = "port";
  if(!ros::param::param<int>(port_parameter, port, default_port))
  {
    ROS_WARN_STREAM_NAMED("init", "Failed to get '" << port_parameter << "' parameter; setting to default value");
  }
  
  const std::string robot_ip_parameter = "robot_ip";
  if(!ros::param::get(robot_ip_parameter, ip) || ip.empty())
  {
    ROS_ERROR_NAMED_STREAM("init", "Invalid IP address; please set the " << robot_ip_parameter << "' parameter");
    return false;
  }
```

Reply:
It would be nice to know when parameters aren't found or a default value is used.

Reply:
You're right, that is nicer. Implemented in 08cbb8e.